openQA.spec

```diff
@@ -48,7 +48,7 @@
 %define assetpack_requires perl(Mojolicious::Plugin::AssetPack) >= 1.36, perl(CSS::Minifier::XS) perl(JavaScript::Minifier::XS)
 %define common_requires perl(Config::IniFiles) perl(Cpanel::JSON::XS) perl(Cwd) perl(Data::Dump) perl(Data::Dumper) perl(Digest::MD5) perl(Getopt::Long) perl(Minion) >= 9.09, perl(Mojolicious) >= 8.24, perl(Try::Tiny) perl(Regexp::Common), perl(Storable)
 # runtime requirements for the main package that are not required by other sub-packages
-%define main_requires %assetpack_requires git-core perl(Carp::Always) perl(Date::Format) perl(DateTime::Format::Pg) perl(DBD::Pg) >= 3.7.4, perl(DBI) >= 1.632, perl(DBIx::Class) >= 0.082801, perl(DBIx::Class::DeploymentHandler) perl(DBIx::Class::DynamicDefault) perl(DBIx::Class::Schema::Config) perl(DBIx::Class::Storage::Statistics) perl(DBIx::Class::OptimisticLocking) perl(File::Copy::Recursive) perl(Net::OpenID::Consumer) perl(Module::Pluggable) perl(aliased) perl(Config::Tiny) perl(Text::Diff) perl(CommonMark) perl(JSON::Validator) perl(IPC::Run) perl(Archive::Extract) perl(Time::ParseDate) perl(Sort::Versions) perl(BSD::Resource) perl(Pod::POM) perl(Mojo::Pg) perl(Mojo::RabbitMQ::Client) >= 0.2, perl(SQL::Translator) perl(YAML::XS) perl(LWP::UserAgent) perl(Getopt::Long::Descriptive)
+%define main_requires %assetpack_requires git-core perl(Carp::Always) perl(Date::Format) perl(DateTime::Format::Pg) perl(DBD::Pg) >= 3.7.4, perl(DBI) >= 1.632, perl(DBIx::Class) >= 0.082801, perl(DBIx::Class::DeploymentHandler) perl(DBIx::Class::DynamicDefault) perl(DBIx::Class::Schema::Config) perl(DBIx::Class::Storage::Statistics) perl(DBIx::Class::OptimisticLocking) perl(File::Copy::Recursive) perl(Net::OpenID::Consumer) perl(Module::Pluggable) perl(aliased) perl(Config::Tiny) perl(Text::Diff) perl(CommonMark) perl(JSON::Validator) perl(IPC::Run) perl(Archive::Extract) perl(Time::ParseDate) perl(Sort::Versions) perl(BSD::Resource) perl(Pod::POM) perl(Mojo::Pg) perl(Mojo::RabbitMQ::Client) >= 0.2, perl(SQL::Translator) perl(YAML::PP) >= 0.020 perl(YAML::XS) perl(LWP::UserAgent) perl(Getopt::Long::Descriptive)
```

Comment:
The only use of YAML::XS seems to be in

```
lib/OpenQA/WebSockets/Controller/Worker.pm:        log_debug(sprintf('Received from worker "%u" worker_status message "%s"', $wid, YAML::XS::Dump($json)));
```

I guess that should be easy to replace and then you can remove YAML::XS as dependency completely in favor of YAML::PP.

Also, https://github.com/os-autoinst/openQA/pull/2759 unfortunately caused a conflict with your changes. You need to rebase and fix that conflict.

Reply:
Conflict resolved

Reply:
Not sure if I should really replace YAML::XS::Dump in Worker.pm.
YAML::XS is much faster than YAML::PP, and this code is called much more often than the jobtemplate generation...

Reply:
Why are we using YAML in the worker at all? It seems to be for a log message? That would be more consistent with `Data::Dumper`.

Reply:
Then better don't replace it.

Reply:
If we are going back to Data::Dumper for the log message, then at least the keys should be sorted. Otherwise I find the log output not helpful.
I find the log output more readable with YAML::XS, and it has sorted keys by default, that's why I changed it originally.

Reply:
`Mojo::Util` contains a function with a very well configured `Data::Dumper` for these kinds of tasks. https://metacpan.org/pod/Mojo::Util#dumper

Reply:
Ok, but do I need to do this as part of this PR?
I find YAML more readable, and it was approved at some point.
Maybe make another PR for using Mojo::Util?

Reply:
Yes, we can merge this PR as soon as tests succeed. I would appreciate if you can followup with the change in another PR so that we don't forget it.

Reply:
> Ok, but do I need to do this as part of this PR?
> I find YAML more readable, and it was approved at some point.
> Maybe make another PR for using Mojo::Util?

I second that concern. The YAML is a lot more useful.